src/core/agent/services/identifierService.ts

```diff
@@ -207,6 +223,19 @@ class IdentifierService extends AgentService {
   }
 
   async deleteStaleLocalIdentifier(identifier: string): Promise<void> {
+    const metadata = await this.identifierStorage.getIdentifierMetadata(
+      identifier
+    );
+
+    if (metadata.groupMetadata) {
```

Comment:
I don't want to delete the linked connections here in case this identifier incorrectly appeared as a stale connection and gets restored. So lets restore this function.